--- 
title: "Protocol for joint genotyping the 3,034 *Saccharomyces cerevisiae* population"
author: "Victor Loegler, Haploteam, Universit√© de Strasbourg, CNRS, GMGM UMR 7156, Strasbourg, France"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/bookdown-demo
description: "This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population. "
---

```{r klippy, echo=FALSE, include=TRUE}
klippy::klippy() # Allow to copy code on click
```

# {-}

```{r, eval=T, echo=F, out.width="100%", fig.align = 'center'}
knitr::include_graphics("3034Tree.jpg", dpi = 200)
```

## Introduction {-}

This book contains the protocol to build the SNP and InDel matrices as for the 3,034 *Saccharomyces cerevisiae* population. The GVCF files of the 3,034 isolates are required and can be downloaded here:

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.12571280.svg)](https://doi.org/10.5281/zenodo.12571280)

Additional isolates can be easily included by generating new GVCF files. Reference genome and bed file required can be found in the zenodo repository. 

This protocol follow the [GATK best practices to identify germline short variants (SNPs and InDels) in one or more individuals to produce a joint callset in VCF format](https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels). 


