<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Generate additional GVCF | Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population</title>
  <meta name="description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Generate additional GVCF | Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Generate additional GVCF | Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population" />
  
  <meta name="twitter:description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  

<meta name="author" content="Victor Loegler, Haploteam, UniversitÃ© de Strasbourg, CNRS, GMGM UMR 7156, Strasbourg, France" />


<meta name="date" content="2024-07-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="joint-genotyping.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Protocol: Sace population joint genotyping</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html"><i class="fa fa-check"></i><b>1</b> Generate additional GVCF</a>
<ul>
<li class="chapter" data-level="1.1" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html#short-reads-mapping"><i class="fa fa-check"></i><b>1.1</b> Short reads mapping</a></li>
<li class="chapter" data-level="1.2" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html#variant-calling"><i class="fa fa-check"></i><b>1.2</b> Variant calling</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="joint-genotyping.html"><a href="joint-genotyping.html"><i class="fa fa-check"></i><b>2</b> Joint genotyping</a>
<ul>
<li class="chapter" data-level="2.1" data-path="joint-genotyping.html"><a href="joint-genotyping.html#consolidate-gvcfs"><i class="fa fa-check"></i><b>2.1</b> Consolidate GVCFs</a></li>
<li class="chapter" data-level="2.2" data-path="joint-genotyping.html"><a href="joint-genotyping.html#joint-call-cohort"><i class="fa fa-check"></i><b>2.2</b> Joint-Call cohort</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="variant-filtering.html"><a href="variant-filtering.html"><i class="fa fa-check"></i><b>3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="3.1" data-path="variant-filtering.html"><a href="variant-filtering.html#quality-filtering"><i class="fa fa-check"></i><b>3.1</b> Quality filtering</a></li>
<li class="chapter" data-level="3.2" data-path="variant-filtering.html"><a href="variant-filtering.html#missing-genotype-filtering"><i class="fa fa-check"></i><b>3.2</b> Missing genotype filtering</a></li>
<li class="chapter" data-level="3.3" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-on-excess-of-heterozygosity"><i class="fa fa-check"></i><b>3.3</b> Filter on excess of heterozygosity</a></li>
<li class="chapter" data-level="3.4" data-path="variant-filtering.html"><a href="variant-filtering.html#split-snps-and-indels"><i class="fa fa-check"></i><b>3.4</b> Split SNPs and InDels</a></li>
<li class="chapter" data-level="3.5" data-path="variant-filtering.html"><a href="variant-filtering.html#discard-non-variant-positions"><i class="fa fa-check"></i><b>3.5</b> Discard non-variant positions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Protocol for joint genotyping the 3,034 <em>Saccharomyces cerevisiae</em> population</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="generate-additional-gvcf" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Generate additional GVCF<a href="generate-additional-gvcf.html#generate-additional-gvcf" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This chapter explains how to generate a GVCF for every additionnal isolate to include in the final variant matrix.</p>
<p>Required softwares:</p>
<ul>
<li>bwa-mem2 (bwa mem can be used, but is slower)</li>
<li>samtools</li>
<li>gatk</li>
</ul>
<p>Commands were successfully run with bwa-mem2 v2.2.1, samtools v1.16.1 and gatk v4.5.0.0.</p>
<div id="short-reads-mapping" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Short reads mapping<a href="generate-additional-gvcf.html#short-reads-mapping" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Short reads mapping have to be done with <code>bwa-mem2 mem</code> (or <code>bwa mem</code>) default parameters. The output of bwa can be piped with <code>samtools sort</code> to output a sorted bam. The <code>-T</code> option allows to give a specific name to temporary files created for the sorting, which allows to sort multiple bam in parallel in the same directory without any name conflict.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="generate-additional-gvcf.html#cb1-1" tabindex="-1"></a><span class="co"># Create reference index</span></span>
<span id="cb1-2"><a href="generate-additional-gvcf.html#cb1-2" tabindex="-1"></a><span class="ex">bwa-mem2</span> index reference.fasta</span>
<span id="cb1-3"><a href="generate-additional-gvcf.html#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="generate-additional-gvcf.html#cb1-4" tabindex="-1"></a><span class="co"># Run mapping</span></span>
<span id="cb1-5"><a href="generate-additional-gvcf.html#cb1-5" tabindex="-1"></a><span class="ex">bwa-mem2</span> mem <span class="at">-t</span> 20 reference.fasta IlluminaReads_1.fq.gz IlluminaReads_2.fq.gz <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">-o</span> Mapping.bam <span class="at">-T</span> reads.Prefix.tmp</span></code></pre></div>
<p>Once the BAM file is generated, one can add ReadGroups (required for some downstream analyses) with <code>gatk AddOrReplaceReadGroups</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="generate-additional-gvcf.html#cb2-1" tabindex="-1"></a><span class="co"># Add read groups</span></span>
<span id="cb2-2"><a href="generate-additional-gvcf.html#cb2-2" tabindex="-1"></a><span class="ex">gatk</span> AddOrReplaceReadGroups <span class="at">-I</span> Mapping.bam <span class="at">-O</span> Mapping.RD.bam <span class="at">--RGID</span> SampleName <span class="at">--RGLB</span> SampleName <span class="at">--RGPL</span> ILLUMINA <span class="at">--RGPU</span> SampleName <span class="at">--RGSM</span> SampleName</span></code></pre></div>
<p>Finally, the bam file has to be indexed with <code>samtools index</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="generate-additional-gvcf.html#cb3-1" tabindex="-1"></a><span class="co"># Index BAM</span></span>
<span id="cb3-2"><a href="generate-additional-gvcf.html#cb3-2" tabindex="-1"></a><span class="ex">samtools</span> index Mapping.RD.bam</span></code></pre></div>
</div>
<div id="variant-calling" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Variant calling<a href="generate-additional-gvcf.html#variant-calling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>HaplotypeCaller is used to call variants and generate a GVCF file from the bam file.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="generate-additional-gvcf.html#cb4-1" tabindex="-1"></a><span class="co"># Index reference genome</span></span>
<span id="cb4-2"><a href="generate-additional-gvcf.html#cb4-2" tabindex="-1"></a><span class="ex">samtools</span> faidx reference.fasta</span>
<span id="cb4-3"><a href="generate-additional-gvcf.html#cb4-3" tabindex="-1"></a><span class="ex">gatk</span> CreateSequenceDictionary <span class="at">-R</span> reference.fasta</span>
<span id="cb4-4"><a href="generate-additional-gvcf.html#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="generate-additional-gvcf.html#cb4-5" tabindex="-1"></a><span class="co"># Index bam file</span></span>
<span id="cb4-6"><a href="generate-additional-gvcf.html#cb4-6" tabindex="-1"></a><span class="ex">samtools</span> index Mapping.RD.bam</span>
<span id="cb4-7"><a href="generate-additional-gvcf.html#cb4-7" tabindex="-1"></a></span>
<span id="cb4-8"><a href="generate-additional-gvcf.html#cb4-8" tabindex="-1"></a><span class="co"># Run HaplotypeCaller</span></span>
<span id="cb4-9"><a href="generate-additional-gvcf.html#cb4-9" tabindex="-1"></a><span class="ex">gatk</span> HaplotypeCaller <span class="at">-R</span> reference.fasta <span class="at">-I</span> Mapping.RD.bam <span class="at">-O</span> Prefix.g.vcf.gz <span class="at">--emit-ref-confidence</span> GVCF</span></code></pre></div>
<p>For sample with really high sequencing depth, variant calling can be performed on each chromosome and subsequently merged into a single GVCF file.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="generate-additional-gvcf.html#cb5-1" tabindex="-1"></a><span class="co"># Run HaplotypeCaller per chromosome</span></span>
<span id="cb5-2"><a href="generate-additional-gvcf.html#cb5-2" tabindex="-1"></a><span class="ex">gatk</span> HaplotypeCaller <span class="at">-R</span> reference.fasta <span class="at">-I</span> Mapping.RD.bam <span class="at">-O</span> Prefix.chromosome1.g.vcf.gz <span class="at">--emit-ref-confidence</span> GVCF <span class="at">-L</span> chromosome1</span>
<span id="cb5-3"><a href="generate-additional-gvcf.html#cb5-3" tabindex="-1"></a><span class="ex">gatk</span> HaplotypeCaller <span class="at">-R</span> reference.fasta <span class="at">-I</span> Mapping.RD.bam <span class="at">-O</span> Prefix.chromosome2.g.vcf.gz <span class="at">--emit-ref-confidence</span> GVCF <span class="at">-L</span> chromosome2</span>
<span id="cb5-4"><a href="generate-additional-gvcf.html#cb5-4" tabindex="-1"></a><span class="ex">gatk</span> HaplotypeCaller <span class="at">-R</span> reference.fasta <span class="at">-I</span> Mapping.RD.bam <span class="at">-O</span> Prefix.chromosome3.g.vcf.gz <span class="at">--emit-ref-confidence</span> GVCF <span class="at">-L</span> chromosome3</span>
<span id="cb5-5"><a href="generate-additional-gvcf.html#cb5-5" tabindex="-1"></a><span class="ex">...</span></span>
<span id="cb5-6"><a href="generate-additional-gvcf.html#cb5-6" tabindex="-1"></a></span>
<span id="cb5-7"><a href="generate-additional-gvcf.html#cb5-7" tabindex="-1"></a><span class="ex">gatk</span> MergeVcfs <span class="at">-I</span> Prefix.chromosome1.g.vcf.gz <span class="at">-I</span> Prefix.chromosome2.g.vcf.gz <span class="at">-I</span> Prefix.chromosome3.g.vcf.gz ... <span class="at">-O</span> Prefix.g.vcf.gz</span></code></pre></div>
<p>Finally, tab-index all newly generated GVCF.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="generate-additional-gvcf.html#cb6-1" tabindex="-1"></a><span class="co"># Index GVCF</span></span>
<span id="cb6-2"><a href="generate-additional-gvcf.html#cb6-2" tabindex="-1"></a><span class="ex">gatk</span> IndexFeatureFile <span class="at">-I</span> Prefix.g.vcf.gz</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="joint-genotyping.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-GenerateAdditionalGVCF.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
