<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population</title>
  <meta name="description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population" />
  
  <meta name="twitter:description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  

<meta name="author" content="Victor Loegler, Haploteam, Université de Strasbourg, CNRS, GMGM UMR 7156, Strasbourg, France" />


<meta name="date" content="2024-07-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="generate-additional-gvcf.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Protocol: Sace population joint genotyping</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html"><i class="fa fa-check"></i><b>1</b> Generate additional GVCF</a>
<ul>
<li class="chapter" data-level="1.1" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html#short-reads-mapping"><i class="fa fa-check"></i><b>1.1</b> Short reads mapping</a></li>
<li class="chapter" data-level="1.2" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html#variant-calling"><i class="fa fa-check"></i><b>1.2</b> Variant calling</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="joint-genotyping.html"><a href="joint-genotyping.html"><i class="fa fa-check"></i><b>2</b> Joint genotyping</a>
<ul>
<li class="chapter" data-level="2.1" data-path="joint-genotyping.html"><a href="joint-genotyping.html#consolidate-gvcfs"><i class="fa fa-check"></i><b>2.1</b> Consolidate GVCFs</a></li>
<li class="chapter" data-level="2.2" data-path="joint-genotyping.html"><a href="joint-genotyping.html#joint-call-cohort"><i class="fa fa-check"></i><b>2.2</b> Joint-Call cohort</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="variant-filtering.html"><a href="variant-filtering.html"><i class="fa fa-check"></i><b>3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="3.1" data-path="variant-filtering.html"><a href="variant-filtering.html#quality-filtering"><i class="fa fa-check"></i><b>3.1</b> Quality filtering</a></li>
<li class="chapter" data-level="3.2" data-path="variant-filtering.html"><a href="variant-filtering.html#missing-genotype-filtering"><i class="fa fa-check"></i><b>3.2</b> Missing genotype filtering</a></li>
<li class="chapter" data-level="3.3" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-on-excess-of-heterozygosity"><i class="fa fa-check"></i><b>3.3</b> Filter on excess of heterozygosity</a></li>
<li class="chapter" data-level="3.4" data-path="variant-filtering.html"><a href="variant-filtering.html#split-snps-and-indels"><i class="fa fa-check"></i><b>3.4</b> Split SNPs and InDels</a></li>
<li class="chapter" data-level="3.5" data-path="variant-filtering.html"><a href="variant-filtering.html#discard-non-variant-positions"><i class="fa fa-check"></i><b>3.5</b> Discard non-variant positions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Protocol for joint genotyping the 3,034 <em>Saccharomyces cerevisiae</em> population</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Protocol for joint genotyping the 3,034 <em>Saccharomyces cerevisiae</em> population</h1>
<p class="author"><em>Victor Loegler, Haploteam, Université de Strasbourg, CNRS, GMGM UMR 7156, Strasbourg, France</em></p>
<p class="date"><em>2024-07-19</em></p>
</div>
<div id="section" class="section level1 unnumbered">
<h1 class="unnumbered"></h1>
<p><img src="3034Tree.jpg" width="100%" style="display: block; margin: auto;" /></p>
<div id="introduction" class="section level2 unnumbered hasAnchor">
<h2>Introduction<a href="index.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This book contains the protocol to build the SNP and InDel matrices as for the 3,034 <em>Saccharomyces cerevisiae</em> population. The GVCF files of the 3,034 isolates are required and can be downloaded here:</p>
<p><a href="https://doi.org/10.5281/zenodo.12571280"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.12571280.svg" alt="DOI" /></a></p>
<p>Additional isolates can be easily included by generating new GVCF files. Reference genome and bed file required can be found in the zenodo repository.</p>
<p>This protocol follow the <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035535932-Germline-short-variant-discovery-SNPs-Indels">GATK best practices to identify germline short variants (SNPs and InDels) in one or more individuals to produce a joint callset in VCF format</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="generate-additional-gvcf.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
