<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Joint genotyping | Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population</title>
  <meta name="description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Joint genotyping | Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Joint genotyping | Protocol for joint genotyping the 3,034 Saccharomyces cerevisiae population" />
  
  <meta name="twitter:description" content="This book contains the protocol to build the SNP and InDel matrices for the 3,034 Saccharomyces cerevisiae population." />
  

<meta name="author" content="Victor Loegler, Haploteam, UniversitÃ© de Strasbourg, CNRS, GMGM UMR 7156, Strasbourg, France" />


<meta name="date" content="2024-07-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="generate-additional-gvcf.html"/>
<link rel="next" href="variant-filtering.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Protocol: Sace population joint genotyping</a></li>

<li class="divider"></li>
<li><a href="index.html#section" id="toc-section"></a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html"><i class="fa fa-check"></i><b>1</b> Generate additional GVCF</a>
<ul>
<li class="chapter" data-level="1.1" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html#short-reads-mapping"><i class="fa fa-check"></i><b>1.1</b> Short reads mapping</a></li>
<li class="chapter" data-level="1.2" data-path="generate-additional-gvcf.html"><a href="generate-additional-gvcf.html#variant-calling"><i class="fa fa-check"></i><b>1.2</b> Variant calling</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="joint-genotyping.html"><a href="joint-genotyping.html"><i class="fa fa-check"></i><b>2</b> Joint genotyping</a>
<ul>
<li class="chapter" data-level="2.1" data-path="joint-genotyping.html"><a href="joint-genotyping.html#consolidate-gvcfs"><i class="fa fa-check"></i><b>2.1</b> Consolidate GVCFs</a></li>
<li class="chapter" data-level="2.2" data-path="joint-genotyping.html"><a href="joint-genotyping.html#joint-call-cohort"><i class="fa fa-check"></i><b>2.2</b> Joint-Call cohort</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="variant-filtering.html"><a href="variant-filtering.html"><i class="fa fa-check"></i><b>3</b> Variant filtering</a>
<ul>
<li class="chapter" data-level="3.1" data-path="variant-filtering.html"><a href="variant-filtering.html#quality-filtering"><i class="fa fa-check"></i><b>3.1</b> Quality filtering</a></li>
<li class="chapter" data-level="3.2" data-path="variant-filtering.html"><a href="variant-filtering.html#missing-genotype-filtering"><i class="fa fa-check"></i><b>3.2</b> Missing genotype filtering</a></li>
<li class="chapter" data-level="3.3" data-path="variant-filtering.html"><a href="variant-filtering.html#filter-on-excess-of-heterozygosity"><i class="fa fa-check"></i><b>3.3</b> Filter on excess of heterozygosity</a></li>
<li class="chapter" data-level="3.4" data-path="variant-filtering.html"><a href="variant-filtering.html#split-snps-and-indels"><i class="fa fa-check"></i><b>3.4</b> Split SNPs and INDELs</a></li>
<li class="chapter" data-level="3.5" data-path="variant-filtering.html"><a href="variant-filtering.html#discard-non-variant-positions"><i class="fa fa-check"></i><b>3.5</b> Discard non-variant positions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Protocol for joint genotyping the 3,034 <em>Saccharomyces cerevisiae</em> population</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="joint-genotyping" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Joint genotyping<a href="joint-genotyping.html#joint-genotyping" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This chapter explains how to jointly genotype all isolates (the 3,034 plus eventual additional isolates), in order to generate a multisample VCF for the whole population.</p>
<p>Required softwares:</p>
<ul>
<li>gatk</li>
</ul>
<p>Commands were successfully run with gatk v4.5.0.0.</p>
<div id="consolidate-gvcfs" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Consolidate GVCFs<a href="joint-genotyping.html#consolidate-gvcfs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>This step consists of consolidating the contents of GVCF files across multiple samples in order to improve scalability and speed the next step, joint genotyping. Note that this is NOT equivalent to the joint genotyping step; variants in the resulting merged GVCF cannot be considered to have been called jointly.</p>
</blockquote>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="joint-genotyping.html#cb1-1" tabindex="-1"></a><span class="co"># Loop on all GVCFs to create a variable that will be given as argument for the GenomicsDBImport function</span></span>
<span id="cb1-2"><a href="joint-genotyping.html#cb1-2" tabindex="-1"></a><span class="va">GVCF_FILES</span><span class="op">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb1-3"><a href="joint-genotyping.html#cb1-3" tabindex="-1"></a><span class="cf">for</span> GVCF <span class="kw">in</span> <span class="pp">*</span>.g.vcf.gz</span>
<span id="cb1-4"><a href="joint-genotyping.html#cb1-4" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb1-5"><a href="joint-genotyping.html#cb1-5" tabindex="-1"></a>  <span class="va">GVCF_FILES</span><span class="op">=</span><span class="va">${GVCF_FILES}</span><span class="st">&quot;-V </span><span class="va">$GVCF</span><span class="st"> &quot;</span></span>
<span id="cb1-6"><a href="joint-genotyping.html#cb1-6" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb1-7"><a href="joint-genotyping.html#cb1-7" tabindex="-1"></a></span>
<span id="cb1-8"><a href="joint-genotyping.html#cb1-8" tabindex="-1"></a><span class="co"># Consolidate GVCFs</span></span>
<span id="cb1-9"><a href="joint-genotyping.html#cb1-9" tabindex="-1"></a><span class="ex">gatk</span> GenomicsDBImport <span class="at">--batch-size</span> 200 <span class="at">--genomicsdb-workspace-path</span> Prefix <span class="at">-L</span> reference.bed <span class="va">$GVCF_FILES</span></span></code></pre></div>
</div>
<div id="joint-call-cohort" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Joint-Call cohort<a href="joint-genotyping.html#joint-call-cohort" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>At this step, we gather all the per-sample GVCFs (or combined GVCFs if we are working with large numbers of samples) and pass them all together to the joint genotyping tool, GenotypeGVCFs. This produces a set of joint-called SNP and indel calls ready for filtering. This cohort-wide analysis empowers sensitive detection of variants even at difficult sites, and produces a squared-off matrix of genotypes that provides information about all sites of interest in all samples considered, which is important for many downstream analyses.</p>
</blockquote>
<p>As some tools require to have VCF files with all positions (even non-variant positions), the <code>--include-non-variant-sites</code> option is used. Non-variant position can be subsequently discarded to have a multi-sample VCF less heavy.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="joint-genotyping.html#cb2-1" tabindex="-1"></a><span class="ex">gatk</span> GenotypeGVCFs <span class="at">--include-non-variant-sites</span> <span class="at">-R</span> reference.fasta <span class="at">-V</span> gendb://Prefix <span class="at">-O</span> Prefix.vcf.gz</span></code></pre></div>
<p>As for the variant calling, this step can be run per chromosome and VCFs can be subsequently merged.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="joint-genotyping.html#cb3-1" tabindex="-1"></a><span class="ex">gatk</span> GenotypeGVCFs <span class="at">--intervals</span> chromosome1 <span class="at">--include-non-variant-sites</span> <span class="at">-R</span> reference.fasta <span class="at">-V</span> gendb://Prefix <span class="at">-O</span> Prefix.chromosome1.vcf.gz</span>
<span id="cb3-2"><a href="joint-genotyping.html#cb3-2" tabindex="-1"></a><span class="ex">gatk</span> GenotypeGVCFs <span class="at">--intervals</span> chromosome2 <span class="at">--include-non-variant-sites</span> <span class="at">-R</span> reference.fasta <span class="at">-V</span> gendb://Prefix <span class="at">-O</span> Prefix.chromosome2.vcf.gz</span>
<span id="cb3-3"><a href="joint-genotyping.html#cb3-3" tabindex="-1"></a><span class="ex">gatk</span> GenotypeGVCFs <span class="at">--intervals</span> chromosome3 <span class="at">--include-non-variant-sites</span> <span class="at">-R</span> reference.fasta <span class="at">-V</span> gendb://Prefix <span class="at">-O</span> Prefix.chromosome3.vcf.gz</span>
<span id="cb3-4"><a href="joint-genotyping.html#cb3-4" tabindex="-1"></a><span class="ex">...</span></span>
<span id="cb3-5"><a href="joint-genotyping.html#cb3-5" tabindex="-1"></a></span>
<span id="cb3-6"><a href="joint-genotyping.html#cb3-6" tabindex="-1"></a><span class="ex">gatk</span> MergeVcfs <span class="at">-I</span> Prefix.chromosome1.vcf.gz <span class="at">-I</span> Prefix.chromosome2.vcf.gz <span class="at">-I</span> Prefix.chromosome3.vcf.gz ... <span class="at">-O</span> Prefix.vcf.gz</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="generate-additional-gvcf.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="variant-filtering.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-JointGenotyping.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
